packages/react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -1489,20 +1489,24 @@ class ComboBoxInternal extends React.Component<IComboBoxInternalProps, IComboBox
 
   private _renderOption = (item: IComboBoxOption): JSX.Element => {
     const { onRenderOption = this._onRenderOptionContent } = this.props;
-    const id = this._id;
+    const id = item.id ?? this._id + '-list' + item.index;
     const isSelected: boolean = this._isOptionSelected(item.index);
     const isChecked: boolean = this._isOptionChecked(item.index);
     const isIndeterminate: boolean = this._isOptionIndeterminate(item.index);
     const optionStyles = this._getCurrentOptionStyles(item);
     const optionClassNames = getComboBoxOptionClassNames(this._getCurrentOptionStyles(item));
     const title = item.title;
 
-    const onRenderCheckboxLabel = () => onRenderOption(item, this._onRenderOptionContent);
+    const onRenderCheckboxLabel = () => (
+      <div id={id + '-label'} aria-hidden="true">
+        onRenderOption(item, this._onRenderOptionContent)
```

Comment:
Does this need `{}`?
```suggestion
        {onRenderOption(item, this._onRenderOptionContent)}
```